src/app-layout/__integ__/app-layout-toolbar-tooltips.test.ts

```diff
@@ -0,0 +1,337 @@
+    return { x, y };
+  }
+
+  async pointerDown(selector: string) {
+    const center = await this.getElementCenter(selector);
+    await (await this.browser.$(selector)).moveTo();
+    await this.browser.performActions([
+      {
+        type: 'pointer',
+        id: 'event',
+        parameters: { pointerType: 'mouse' },
+        actions: [
+          { type: 'pointerMove', duration: 0, origin: 'pointer', ...center },
+          { type: 'pointerDown', button: 0 },
+          { type: 'pause', duration: 100 },
+        ],
+      },
+    ]);
+  }
+
+  async pointerUp() {
+    await this.browser.performActions([
+      {
+        type: 'pointer',
+        id: 'event',
+        parameters: { pointerType: 'mouse' },
+        actions: [
+          { type: 'pointerUp', button: 0 },
+          { type: 'pause', duration: 100 },
+        ],
+      },
+    ]);
+  }
+}
+
+interface SetupTestOptions {
+  splitPanelPosition?: string;
+  size?: 'desktop' | 'mobile';
+  disableContentPaddings?: string;
+  visualRefresh?: string;
+}
+
+const drawerIds = Object.values(drawerIdObj);
+const VISIBLE_MOBILE_TOOLBAR_TRIGGERS_LIMIT = 2; //must match the number in  '../../../lib/components/app-layout/visual-refresh/drawers';
+const mobileDrawerTriggerIds = drawerIds.slice(0, VISIBLE_MOBILE_TOOLBAR_TRIGGERS_LIMIT);
+
+const setupTest = (
+  {
+    splitPanelPosition = 'bottom',
+    size = 'desktop',
+    disableContentPaddings = 'false',
+    visualRefresh = 'false',
+  }: SetupTestOptions,
+  testFn: (page: AppLayoutDrawersPage) => Promise<void>
+) =>
+  useBrowser(size === 'desktop' ? viewports.desktop : viewports.mobile, async browser => {
+    const page = new AppLayoutDrawersPage(browser);
+    const params = new URLSearchParams({
+      visualRefresh,
+      splitPanelPosition,
+      disableContentPaddings,
+    }).toString();
+    await browser.url(`#/light/app-layout/with-drawers?${params}`);
+    await page.waitForVisible(wrapper.findContentRegion().toSelector());
+    await testFn(page);
+  });
+
+describe(`theme='visual-refresh'`, () => {
+  describe(`desktop`, () => {
+    const size = 'desktop';
+
+    test(
+      'Shows tooltip correctly for mouse interactions on desktop',
+      setupTest({ disableContentPaddings: 'true', visualRefresh: 'true', size }, async page => {
+        await expect(page.isExisting(`.${visualRefreshStyles['trigger-tooltip']}`)).resolves.toBeFalsy();
+        await expect(
+          page.isExisting(`.${visualRefreshStyles[`drawers-desktop-triggers-container`]}`)
+        ).resolves.toBeTruthy();
+        await expect(page.isExisting(`.${visualRefreshStyles['drawers-trigger-overflow']}`)).resolves.toBeFalsy();
+        await page.hoverElement(wrapper.findDrawerTriggerById(drawerIds[0]).toSelector());
+        await expect(page.isExisting(`.${visualRefreshStyles['trigger-tooltip']}`)).resolves.toBeTruthy();
+
+        for (const drawerId of drawerIds) {
+          async () => {
```

Comment:
Doesn't this `async () { ... }` create a function? Is the block inside actually called? Same in line 120

Reply:
Fixed this. But the tests were timing out so I had to simplify them